### PR TITLE
Compactor support time filter (#4884)

### PR DESCRIPTION
**Pull request by jordy1024**
* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

### CODE REVIEW

**Comment by metonymic-smokey** [comment]
Looks good to me(I'm not an official reviewer though).
Maybe just check once if the compact tests require any additions, since some of them invoke the other filters in compact so I think this should be added to those tests too?

**Comment by metonymic-smokey** [comment]
@jordy1024 also to fix the failing docs test, please run `make docs` locally and push the resulting changes.

**Comment by jordy1024** [comment]
> @jordy1024 also to fix the failing docs test, please run `make docs` locally and push the resulting changes.

@metonymic-smokey  Thanks for reminding me,I will fix it right away.

**Comment by jordy1024** [comment]
Hello,metonymic-smokey @metonymic-smokey 
what should I do?
The CI process is blocked every time in the  `Documentation check`  step, and the error message shows  
`error: 		mixin/README.md:186: "https://www.prometheus.io/docs/prometheus/latest/configuration/unit_testing_rules/" not accessible; status code 400: Bad Request` , but the official prometheus document link is actually accessible.

**Comment by metonymic-smokey** [comment]
@jordy1024 I faced the same issue in one of my PRs today.
As far as I understand, since that file is not related to your changes, the only way out is to maybe have a maintainer re-trigger the tests.

**Comment by jordy1024** [comment]
@metonymic-smokey 
I think so too, thank you very much.

**Comment by jordy1024** [comment]
@metonymic-smokey 
Thank you for helping me review the code;
Can you tell me your contact information, such as email. This may be more convenient for instant communication.

**Comment by metonymic-smokey** [comment]
@jordy1024 np at all! 
You can find me on the #thanos slack as 'Aditi'.

File: cmd/thanos/compact.go
Comment (yeya24) [inline cmd/thanos/compact.go:746]: Please update the description `Thanos Compactor will serve only metrics`. `serve` is not a suitable verb for compactor.

File: cmd/thanos/compact.go
Comment (yeya24) [inline cmd/thanos/compact.go:749]: ditto

**Comment by jordy1024** [comment]
@yeya24 Maybe it's better, use compact instead of serve and blocks instead of metrics.
Please help to review again and unblock CI.

**Comment by jordy1024** [comment]
I'm not sure if it is my problem. An error occurred in Phase `go / Thanos end-to-end tests (pull_request)`: XXX; 
```
manifest for quay.io/prometheus/busybox@sha256:de4af55df1f648a334e16437c550a2907e0aed4f0b0edf454b0b215a9349bdbb not found: manifest unknown: manifest unknown
38
Error: make: *** [Makefile:155: docker] Error 1
39
Error: Process completed with exit code 2.
```
what should I do?

**Comment by jordy1024** [comment]
I'm not sure if it is my problem. An error occurred in Phase `go / Thanos end-to-end tests (pull_request)`: 

```
manifest for quay.io/prometheus/busybox@sha256:de4af55df1f648a334e16437c550a2907e0aed4f0b0edf454b0b215a9349bdbb not found: manifest unknown: manifest unknown
Error: make: *** [Makefile:155: docker] Error 1
Error: Process completed with exit code 2.
```

what should I do?

**Comment by yeya24** [comment]
> I'm not sure if it is my problem. An error occurred in Phase `go / Thanos end-to-end tests (pull_request)`:
> 
> ```
> manifest for quay.io/prometheus/busybox@sha256:de4af55df1f648a334e16437c550a2907e0aed4f0b0edf454b0b215a9349bdbb not found: manifest unknown: manifest unknown
> Error: make: *** [Makefile:155: docker] Error 1
> Error: Process completed with exit code 2.
> ```
> 
> what should I do?

Can you please do a rebase on main branch? I think the fix is on the main branch already

**Review by yeya24 — DISMISSED** [review]
@jordy1024 This looks good now. Can you please update the changelog?

**Comment by jordy1024** [comment]
> changelog



> changelog



> @jordy1024 This looks good now. Can you please update the changelog?

**Comment by jordy1024** [comment]
> @jordy1024 This looks good now. Can you please update the changelog?

Sorry,it was an oversight on my part. I will complete it right away. @yeya24

**Comment by jordy1024** [comment]
@yeya24 
Sorry, because I misoperation of `git rebase` caused some confusion and the current PR was closed. How can I reopen the current PR?

**Review by yeya24 — APPROVED** [review]
LGTM

**Pull request merged**
